posts/2022-12-14-android-ndk-update.md

```diff
@@ -0,0 +1,25 @@
+---
+layout: post
+title: "Updating the Android NDK in Rust <version>"
+author: Android Platform Team
+description: "Modernizing Android support in Rust"
+---
+
+We are pleased to announce that Android platform support in Rust will be
+modernized in Rust <version> as we update the target NDK from r17 to r25.  As a
+consequence the minimum supported API level will increase from 15 (Ice Cream
+Sandwich) to 19 (KitKat).
+
+In NDK r23 Android switched to using LLVM's `libunwind` for all architectures.
+This meant that
+1. If a project were to target NDK r23 or newer with previous versions of Rust
+   [a workaround](https://github.com/rust-lang/rust/pull/85806#issuecomment-1096266946)
+   would be required to redirect attempts to link against `libgcc` to instead
+   link against `libunwind`.  Following this update this workaround will no
+   longer be necessary.
+2. If a project uses NDK r22 or older it will need to be updated to use r23 or
```

Comment:
Is there anything we can link to that's an explainer for how to do this? Do we expect anyone targeting Android to know what they're currently using, or would it be possible to give some guidance (e.g., "run this command to check what you're targeting today").

Reply:
I believe users should know how to do this.  They would have had to have installed the NDK (either through Studio or by downloading it manually) and then passed that path into their Rust build system to use it at all, so presumably they've already accomplished these steps at least once.

And if they ended up needing to apply this workaround they should be acutely aware of what version of the NDK they are using.

Reply:
At least once a few years back (e.g., for an app that is only partially maintained) seems pretty likely to at the very least be out of cache.

Is it hard to link to *something* here? Even the page for NDK downloads or Studio is probably of some help to users.

Reply:
I've added a link to a download page and the NDK's page on integrating the toolchain with non-CMake build systems.  I can't think of any other helpful links off the top of my head but I'd be happy to add any more that we come up with.

Reply:
Skimming that page it looks great to me.